src/select.rs

```diff
@@ -104,25 +140,85 @@ impl<T: Clone> PromptInteraction<T> for Select<T> {
+
+        let items_display: String;
+        let footer_display = theme.format_footer(&state.into());
 
-        let line2: String = self
-            .items
-            .iter()
-            .enumerate()
-            .map(|(i, item)| {
-                theme.format_select_item(&state.into(), self.cursor == i, &item.label, &item.hint)
-            })
-            .collect();
-        let line3 = theme.format_footer(&state.into());
+        if self.enable_filter_mode && !self.input_filter.is_empty() {
+            let mut display = String::new();
+            display += &header_display;
+
+            let mut filtered_and_scored_items: Vec<_> = self
+                .items
+                .iter()
+                .map(|item| {
+                    let similarity = strsim::jaro_winkler(
+                        &item.label.to_lowercase(),
+                        &self.input_filter.value_to_string().to_lowercase(),
+                    );
+                    (similarity, item)
+                })
+                .filter(|(similarity, _)| *similarity > 0.6)
+                .collect();
+
+            filtered_and_scored_items.sort_by(|a, b| b.0.partial_cmp(&a.0).unwrap());
+
+            self.filtered_items = filtered_and_scored_items
+                .into_iter()
+                .map(|(_, item)| item.clone())
+                .collect();
+
+            if !self.filtered_items.is_empty() && self.cursor > self.filtered_items.len() - 1 {
+                self.cursor = 0;
+            }
+
+            let filter_display = theme.format_input(&state.into(), &self.input_filter);
+
+            let items_display: String = self
+                .filtered_items
+                .iter()
+                .enumerate()
+                .map(|(i, item)| {
+                    theme.format_select_item(
+                        &state.into(),
+                        self.cursor == i,
+                        &item.label,
+                        &item.hint,
+                    )
+                })
+                .collect();
+
+            display += &filter_display;
+            display += &items_display;
+            display += &footer_display;
+            display
+        } else {
+            items_display = self
+                .items
+                .iter()
+                .enumerate()
+                .map(|(i, item)| {
+                    theme.format_select_item(
+                        &state.into(),
+                        self.cursor == i,
+                        &item.label,
+                        &item.hint,
+                    )
+                })
+                .collect();
+            header_display + &items_display + &footer_display
+        }
+    }
 
-        line1 + &line2 + &line3
+    fn input(&mut self) -> Option<&mut StringCursor> {
+        Some(&mut self.input_filter)
     }
 }
 
 #[cfg(test)]
 mod test {
-    use super::*;
+    use crate::Select;
```

Comment:
In Rust, `use super::*` is allowed in unit tests and is idiomatic for unit tests :)